packages/principal/src/index.test.ts

```diff
@@ -42,3 +43,35 @@ describe('Principal', () => {
     expect(anonymous.compareTo(principal2)).toBe('gt');
   });
 });
+
+describe('backwards compatibility with 0.9.3', () => {
+  test('import from 0.9.3 principal', () => {
+    const testPrincipal = PrincipalLegacy.anonymous();
+    expect(Principal.from(testPrincipal)).toMatchInlineSnapshot(`
+      Principal {
+        "_arr": Uint8Array [
+          4,
+        ],
+        "_isPrincipal": true,
+      }
+    `);
+  });
+  test('alignment on resolution', () => {
+    expect(PrincipalLegacy.fromHex('efcdab000000000001').toText()).toBe(
+      '2chl6-4hpzw-vqaaa-aaaaa-c',
+    );
+    const anonymous = Principal.anonymous();
+    const anonymousLegacy = Principal.anonymous();
+    expect(anonymous.toText()).toMatch(anonymousLegacy.toText());
+
+    const principal1 = Principal.fromText('ryjl3-tyaaa-aaaaa-aaaba-cai');
+    const principal2 = PrincipalLegacy.fromText('ivg37-qiaaa-aaaab-aaaga-cai');
+
+    expect(principal1.compareTo(principal2)).toBe('lt');
+  });
+  test('assignment to PrincipalBase', () => {
+    const current = Principal.anonymous() as PrincipalBase;
```

Comment:
What is the purpose of `current` in this test?  (Wondering about the warnings)

Reply:
It's to verify that the assignment - `as PrincipalBase` doesn't throw TypeScript errors. If the signature was incompatible, it would mean that the abstract class is no longer compatible with the current version